common/scripts/mesh-helper.sh

```diff
@@ -43,21 +43,34 @@ EOF
   fi
 }
 
-generate_br_lan_ip() {
+generate_bridge_ip() {
   local mesh_if_mac
-  mesh_if_mac="$(ip -brief link | grep "$MESH_VIF" | awk '{print $3; exit}')"
+
+  bridge_name=$(echo "$BRIDGE" | cut -d' ' -f1)
+
+  mesh_if_mac="$(ip -brief link | grep "${id0_MESH_VIF}" | awk '{print $3; exit}')"
+  if [ -z "$mesh_if_mac" ]; then
+      echo "generate_bridge_ip: MAC not found for id0_MESH_VIF! Configuration error?" > /dev/kmsg
+      mesh_if_mac="$(ip -brief link | grep "eth0" | awk '{print $3; exit}')"
+  fi
   local ip_random
   ip_random="$(echo "$mesh_if_mac" | cut -b 16-17)"
```

Comment:
As reported before but maybe not noticed yet: **IPv4 CANNOT BE SELF-CONFIGURED** :)

For example:

- If the MAC address ends in 00 then the bridge will try to use 192.168.1.0 i.e. the reserved network address in this /24
- If the MAC address ends in ff then the bridge will try to use 192.168.1.255 i.e. the reserved broadcast address in this /24
- It takes only 19 CommsDevices for "two of them picked the same IP" to happen more often than not.

**Please stop using IPv4** wherever possible :) For example here
- the interface most likely already has an IPv6 LLA (Link Local Address, the kind that starts with fe80)
- which you can change to one derived from one of the CommsDevice's physical MAC addresses if that is not what it did already
- or enable Duplicate Address Detection (not the best)
- or give it an IPv6 Unique Local Address with the first half `fd[something chosen randomly but once forever]` and the second half an EUI-64 from one of the CommsDevice's physical MAC addresses for example 00:20:91:ab:cd:ef would become 0220:91ff:fe:cdef (simple algorithm that inserts ff:fe in the middle and flips the Locally Administered bit).

Reply:
This is not a valid comment for multiradio implementation.  We are re-using a code which was done much earlier.   IPv4 to IPv6 change was not in a scope of multiradio work.

Reply:
Sorry I am new to PR comments, I was hoping this is an opportunity to clean up the "going to waste hours in debug" broken IPv4 code.

Reply:
ok but what do we do for now about this not being able to work if someone ends up having a MAC address ending in 00 or FF for example? We blame hardware and send another device? 🙂

Reply:
I suppose IPv6 should fix this or is there bottleneck in slaac design?

Reply:
Resolving this as IPv6 was not in scope.